pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
ok ... @bparees and I talked, on same page ... a k8s pull to either make the k8s version image not found error type public, or allow for a way to inject this private error not found into fake client, is desired.